### PR TITLE
Fix #28 Added mission date filter

### LINKED ISSUE #28: Search Mission by creation date
Search Mission by creation date

### PR DESCRIPTION
**Pull request by MV88**
issue #28

Added the mission date filter for missions list

### CODE REVIEW

File: js/components/mission/MissionList.jsx [hunk 1]
```diff
@@ -35,9 +34,11 @@ class MissionList extends React.Component {
     static propTypes = {
         missions: PropTypes.array,
         assets: PropTypes.array,
+        assetCurrent: PropTypes.object,
```
Comment (mbarto) [inline js/components/mission/MissionList.jsx]: currentAsset

File: js/components/mission/MissionDateFilter.jsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+*/
+
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import DateField from '@mapstore/components/data/query/DateField';
+// import Message from '@mapstore/components/I18N/Message';
```
Comment (mbarto) [inline js/components/mission/MissionDateFilter.jsx]: Remove

File: js/components/mission/MissionList.jsx [hunk 3]
```diff
@@ -47,12 +48,13 @@ class MissionList extends React.Component {
         loadingMissions: false,
         missions: [],
         onChangeCurrentMission: () => {},
-        onSelectMission: () => {}
+        onSelectMission: () => {},
+        renderDateFilter: () => null
```
Comment (mbarto) [inline js/components/mission/MissionList.jsx]: dateFilterComponent?

File: js/plugins/index.js [hunk 4]
```diff
@@ -46,15 +48,19 @@ import {
...
+    updateDateFilterValue,
+    updateDateFilterException,
     updateDroneGeometry,
     zoomToItem
 } from '@js/actions/sciadro';
 
 import {
     anomaliesListSelector,
     assetsListSelector,
+    assetCurrentSelector,
```
Comment (mbarto) [inline js/plugins/index.js]: currentAssetSelector

File: js/reducers/sciadro.js [hunk 5]
```diff
@@ -276,6 +294,31 @@ export default function sciadro(state = {
...
             };
         }
+        case FILTER_MISSION_BY_DATE: {
+            const startDate = new Date(state.dateFilter.fieldValue.startDate);
+            const endDate = new Date(state.dateFilter.fieldValue.endDate);
+            switch (state.dateFilter.operator) {
+                case "><": {
+                    return {
+                        ...state,
+                        missionDateFilter: mission => {
```
Comment (mbarto) [inline js/reducers/sciadro.js]: I would move this missionDateFilter functions in a selector, I don't like putting functions in the state.
The simpler is the state, the better it is.

Comment (MV88) [reply js/reducers/sciadro.js]: yes makes sense,
in that case i need a flag that changes the way missions are filtered and not every time the date value changes (it is so by design, at least for the moment )

**Pull request merged**